conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,600 @@
+                                // @error missing identifier
+    io:println(let i = 2 in i); // @error unknown type 'i'
+                                // @error missing identifier
+                                // @error undefined symbol 'i'
+    io:println(let int i = 2 i); // @error missing in keyword
+    io:println(let int i = 2 int j = 2 in i + j); // @error missing comma token
+    io:println(let in i); // @error missing let variable declaration
+                          // @error undefined symbol 'i'
+    io:println(let i); // @error unknown type 'i'
+                       // @error missing equal token
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing in keyword
+}
+
+Test-Case: error
+Description: let-expr when variables are declared outside its' scope.
+Labels: let-expr, int
+
+function errorFunction() {
+    int i = 2;
+    io:println(let int i = 2 in i); // @error redeclared symbol 'i'
+}
+
+Test-Case: error
+Description: let-expr when the type of let-var-decl does not match
+Labels: let-expr
+
+function errorFunction() {
+    io:println(let string i = "2", int j = 3 in i + j); // @error operator '+' not defined for 'string' and 'int'
+    io:println(let int i = 2 in foo(i)); // @error incompatible types: expected 'string?', found 'int'
+    io:println(let var i = 2 in foo(i)); // @error incompatible types: expected 'string?', found 'int'
+}
+
+function foo(string? s) returns string? {
+    return s;
+}
+
+Test-Case: output
+Description: Test let-expr being an isolated expression, when the expression following `in` is isolated.
+             Referenced variables bound by let-var-decls are initialized using isolated expressions.
+Labels: let-expr, function-call-expr, int
+
+isolated int x = 2;
+isolated int y = 4;
+
+isolated function bar(int i) returns int {
+    return i;
+}
+
+isolated function init() {
+    lock {
+        x = let int i = 20 in i;
+        io:println(x); // @output 20
+        x = let int i = 22 in bar(i);
+        io:println(x); // @output 22
+    }
+
+    lock {
+        int x1 = let int i = y in bar(i);
+        io:println(x1); // @output 4
+        int x2 = let int i = bar(200) in bar(i);
+        io:println(x2); // @output 200
+    }
+}
+
+Test-Case: error
+Description: Test error cases when let-expr is assigned to an isolated expression.
+             Non isolated expressions cannot be used to initialize let-var-decl when expression following in is isolated.
+Labels: let-expr, function-call-expr, array-type
+
+isolated int[] x = [];
+
+function foo() {
+    lock {
+        int[] y = [23];
+        x = let int[] i = y in i; // @error invalid attempt to transfer out a value from a 'lock' statement with restricted variable usage: expected an isolated expression and invalid attempt to transfer a value into a 'lock' statement with restricted variable usage
```

Comment:
Please simplify this for the let expr context.

Reply:
Changed